cmd/performance-profile-creator/cmd/root.go

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
+
+	"k8s.io/klog"
+	"k8s.io/utils/pointer"
+
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	powerConsumptionMode, outputFilePath, mustGatherDirPath, profileName string
+	reservedCPUCount                                                     int
+	splitCPUsAcrossNUMA, disableHT, rtKernel, userLevelNetworking        bool
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "performance-profile-creator",
+	Short: "A tool that automates creation of Performance Profiles",
+	Run:   func(cmd *cobra.Command, args []string) {},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.PersistentFlags().IntVarP(&reservedCPUCount, "reserved-cpu-count", "R", 0, "Number of reserved CPUs")
+	rootCmd.PersistentFlags().BoolVarP(&splitCPUsAcrossNUMA, "split-cpus-across-numa", "S", true, "Split the CPUs across NUMA nodes")
+	rootCmd.PersistentFlags().BoolVarP(&disableHT, "disable-ht", "H", false, "Disable Hyperthreading")
+	rootCmd.PersistentFlags().BoolVarP(&rtKernel, "rt-kernel", "K", true, "Enable Real Time Kernel")
+	rootCmd.PersistentFlags().BoolVarP(&userLevelNetworking, "user-level-networking", "U", false, "Run with User level Networking(DPDK) enabled")
+	rootCmd.PersistentFlags().StringVarP(&powerConsumptionMode, "power-consumption-mode", "P", "cstate", "The power consumption mode")
+	rootCmd.PersistentFlags().StringVarP(&outputFilePath, "output-file-path", "O", "/output/sample-profile.json", "Performance Profile file output path")
```

Comment:
no absolute path please, always relative unless there is a really strong reason to use absolute (there is?)

Reply:
We need to be able to extract the output perf-prof yaml file after the tool outputs it to some dir.
We are open to suggestions

Reply:
Which is fine, but why the default path needs to be absolute (and why the subdirectory btw?) 
IOW, can't we just use `sample-profile.json` as default?

Reply:
I am not sure we can, /usr/local/bin is ro.
The /output aims to be a designated directory for all the artifacts. We use a tmpfs volume mount to make it rw
Any other wayto fo it?

Reply:
But maybe we can start sending the output to STDOUT? No arg, no mount?

Reply:
well, if we want to set defaults to be super-specific to the containerized run use case and thus to make the podman invocation (somehow) shorter, well is not a big deal and we can move forward.
BTW, if the profile will be sent to stdout (see comment below) which output do we expect?

Reply:
Nothing for now, let's send to STDOUT for now, we will need to send some debug info later on. 
But let's start simple.

Reply:
Sure, will update to send the output to STDOUT.

Reply:
Debug only on stderr please. (all the logging packages already do that by default). Never pollute stdout with non-output (e.g. logging) data.

Reply:
We were thinking of running the tool something like :

Reply:
But the profile IS the data
The debug message will go to stderr

Reply:
We send the profile to stdout, all debug /logs to stderr
So we can run:  podman run --entrypoint performance-profile-creator  -v ./must-gather:/must-gather:z quay.io/swsehgal/perf-profile-creator:latest > my-profile.yaml
Then
oc apply -f my-profile.yaml